components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -0,0 +1,364 @@
+
+    private static final Log log = LogFactory.getLog(SecretManagerImpl.class);
+    private final List<SecretDAO> secretDAOS;
+
+    public SecretManagerImpl() {
+
+        this.secretDAOS = SecretManagerComponentDataHolder.getInstance().getSecretDAOS();
+    }
+
+    @Override
+    public Secret addSecret(Secret secret) throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        validateSecretCreateRequest(secret);
+        String secretId = generateUniqueID();
+        secret.setSecretId(secretId);
+        secret.setSecretValue(getEncryptedSecret(secret.getSecretValue(), secret.getSecretName()));
+        this.getSecretDAO().addSecret(secret);
+        if (log.isDebugEnabled()) {
+            log.debug("Secret: " + secret.getSecretName() + " added successfully");
+        }
+        return secret;
+    }
+
+    @Override
+    public Secret getSecret(String secretName) throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        validateSecretRetrieveRequest(secretName);
+        Secret secret = this.getSecretDAO().getSecretByName(secretName, getTenantId());
+        if (secret == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret found for the secretName: " + secretName);
+            }
+            throw handleClientException(ERROR_CODE_SECRET_DOES_NOT_EXISTS, secretName, null);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Secret: " + secretName + " is retrieved successfully.");
+        }
+        return secret;
+    }
+
+    @Override
+    public Secrets getSecrets() throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        List secretList = this.getSecretDAO().getSecrets(getTenantId());
+        if (secretList == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret found for the tenant: " + getTenantDomain());
+            }
+            throw handleClientException(
+                    SecretConstants.ErrorMessages.ERROR_CODE_SECRETS_DOES_NOT_EXISTS, null);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("All secrets of tenant: " + getTenantDomain() + " are retrieved successfully.");
+        }
+        return new Secrets(secretList);
+    }
+
+    @Override
+    public Secret getSecretById(String secretId) throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        if (StringUtils.isBlank(secretId)) {
+            throw handleClientException(ERROR_CODE_INVALID_SECRET_ID, secretId);
+        }
+        Secret secret = this.getSecretDAO().getSecretById(secretId, getTenantId());
+        if (secret == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret found for the secretId: " + secretId);
+            }
+            throw handleClientException(SecretConstants.ErrorMessages.ERROR_CODE_SECRET_ID_DOES_NOT_EXISTS, secretId);
```

Comment:
Shall we import the error message constant here as well, since you have imported in other places to make it consistent?
Fix in similar places.

Reply:
Addressed with the commit [4191a12](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/4191a12ce26a59b9a12028b142a6964aeb4ddd2a)